src/components/encoding-pane/property-editorSchema.ts

```diff
@@ -0,0 +1,79 @@
+import * as vlSchema from 'vega-lite/build/vega-lite-schema.json';
+
+export const AXIS_TITLE_SCHEMA = {
+  "type": "object",
+  "title": "Axis",
+  "properties": {
+    "axisTitle": {
+      "type": "string",
+      "title": "Title"
+    },
+  },
+};
+
+export const STACK_SCHEMA = {
+  "type": "object",
+  "title": "Stack",
+  "properties": {
+    "stackSelect": {
+      "title": "Offset",
+      "enum": (vlSchema as any).definitions.StackOffset.enum,
+      "type": "string"
+    }
+  }
+};
+
+export const AXIS_ORIENT_SCHEMA = {
+  "type": "object",
+  "title": "Axis",
+  "properties": {
+    "orient": {
+      "title": "Orient",
+      "enum": (vlSchema as any).definitions.AxisOrient.enum,
+      "type": "string"
+    }
+  }
+};
+
+export const SCALE_TYPE_SCHEMA = {
+  "type": "object",
+  "title": "Scale",
+  "properties": {
+    "scaleTypeSelect": {
+      "title": "Type",
+      "enum": (vlSchema as any).definitions.ScaleType.enum,
+      "type": "string"
+    },
+  },
+};
+
+export const AXIS_TITLE_UISCHEMA = {
```

Comment:
Why are XXX_UISCHEMA declared separately from XXX_SCHEMA?

Reply:
I also would like to make sure all these constants have proper interfaces so they are consistently declared.

Reply:
Plus, I see that the key in the SCHEMA and UISCHEMA are always the same, but given you have to type them separately, this can be error prone. 

Better declare a factory that takes: 
1) parent title 
2) property name (if there is always only one property for each declared schema) 
3) schema (title, enum, type)
4) uiSchema.  

and consistently generate it. 


or even better replace property name with nestedProp and always generate a consistent name pattern.